### PR TITLE
Refreshes user settings list when flag state changes.

### PR DESCRIPTION
**Pull request by dlochrie**
## Hey, I just made a Pull Request!

Updates feature flags settings page so that the flags displayed change when the flag state changes.

Reflects the same state when the page is refreshed.

https://github.com/backstage/backstage/assets/1228728/138abce4-74fd-4a2c-8c5b-a892683083cf

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx [hunk 1]
```diff
@@ -77,6 +77,13 @@ export const UserSettingsFeatureFlags = () => {
         ...prevState,
         [flagName]: newState === FeatureFlagState.Active,
       }));
+
+      const updatedFlags = sortFlags(
+        featureFlagsApi.getRegisteredFlags(),
+        featureFlagsApi,
+      );
+
+      setFeatureFlags(updatedFlags);
```
Comment (Rugvip) [inline plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx:86]: Skipping this step is actually by design, we don't want the flags to re-order while making changes as it's quite jarring. The flag value still gets updated under the hood though, the `featureFlags` state here only represents the ordering of the flags, not whether they're active or not, that's the `featureFlagsApi.save()` call above

File: plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx [hunk 2]
```diff
@@ -99,9 +106,6 @@ export const UserSettingsFeatureFlags = () => {
     <Grid container style={{ justifyContent: 'space-between' }}>
       <Grid item xs={6} md={8}>
         <Typography variant="h5">Feature Flags</Typography>
-        <Typography variant="subtitle1">
-          Please refresh the page when toggling feature flags
-        </Typography>
```
Comment (Rugvip) [inline plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx:104]: The feature flag system is not built to support runtime changes to feature flags. There might be some instances where it works, but often a full page reload is needed. To implement this correctly we'd need some form of metadata attached to the flag that indicates whether it supports runtime changes or not.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you for the suggestion! 👍 

Unfortunately this is a bit trickier to implement correctly, but let us know if you want to take a stab at that path

File: plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx (hunk 2 above)
Comment (dlochrie) [reply plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx:104]: Ah, that makes sense. Might need to do some digging to find a way to surface that metadata to achieve this in a way that provides nice feedback to user, but also takes runtime/!runtime flags into consideration.

**Comment by dlochrie** [comment]
Closing, will come back to this when time allows to make this work for different types of flags.

**Pull request closed without merging**